### PR TITLE
Improve checks on pull

### LINKED ISSUE #422: "Input must be planar" error stopping pull entirely
#### Description:

When pulling panels from Robot, the "Input must be planar" error message will stop the pull entirely, even though most of the panels in the Robot model are completely fine. Although this issue can theoretically be worked around by fixing the problem in the Robot model, finding out which panel is causing this issue would be incredibly tedious.

#### Steps to reproduce:

Using any request involving panels, such as a standard request for the Panel type or a MeshResultRequest, on an affected Robot model.

#### Expected behaviour:
Ideally, it would output any panels it can and skip those causing this error, rather than the entire pull failing completely as it does now. An alternative might be to be more transparent about which panel is causing the error, so that can be fixed in the Robot model.

#### Test file(s):

Not publicly attaching any test files at the moment. Contact me if you need one.

### LINKED ISSUE #431: Robot_Toolkit: Improve null and other checking on pulling from robot
#### Description:

Better nullchecks and other checks to be implemented.

Aim is to have the adapter not crash, but return all that it can for the given query and give less cryptic error messages for the rest.

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

 Closes #431 
Closes #422 

 

Adding multiple checks on Read methods. Should work as before, with no crashes on models with errors in them.

Have tried to add a significant amount of checks, but might still have missed something, but should be significantly more stable now.

 ### Test files

 ### Changelog

 ### Additional comments

### CODE REVIEW

**Review by IsakLarbornBH — COMMENTED** [review]
This solves most of the issues I've been having with pulling panels from Robot. However, using a MeshResultRequest still crashes with no data being returned, showing the same "Input must be planar" error as before. 

This is a vast improvement though, and the rest of the adapters deserve the same treatment. Any data is better than no data.

**Comment by IsakNaslundBh** [comment]
> This solves most of the issues I've been having with pulling panels from Robot. However, using a MeshResultRequest still crashes with no data being returned, showing the same "Input must be planar" error as before.
> 
> This is a vast improvement though, and the rest of the adapters deserve the same treatment. Any data is better than no data.

Thanks @IsakLarbornBH . Will have a look into the mesh result issues you describe, and try to get them working as well! Assume it is the same model as before?

**Comment by IsakNaslundBh** [comment]
Have added some more checks and protection for reading the mesh results now @IsakLarbornBH Think it should be working as expected now!

**Review by IsakLarbornBH — COMMENTED** [review]
Works perfectly for all my use cases now. Thanks!

One thing that crossed my mind is that the error message (Input must be planar) could be a bit misleading now as it implies a strict requirement, but the panels are being let through anyway. Might be a good idea to reword it to something along the lines of "Input is non-planar".

**Comment by IsakNaslundBh** [comment]
> Works perfectly for all my use cases now. Thanks!
> 
> One thing that crossed my mind is that the error message (Input must be planar) could be a bit misleading now as it implies a strict requirement, but the panels are being let through anyway. Might be a good idea to reword it to something along the lines of "Input is non-planar".

That error message is actually coming from the Geometry_Engine, when called by the Structure_Engine, trying to extract the normal of the Panel. Agree this comment is not that helpful, and should go, but that needs to be tackled in here https://github.com/BHoM/BHoM_Engine/issues/2109

**Review by IsakLarbornBH — APPROVED** [review]
> Agree this comment is not that helpful, and should go, but that needs to be tackled in here [BHoM/BHoM_Engine#2109](https://github.com/BHoM/BHoM_Engine/issues/2109)

And would you look who is assigned to that one. Well, now I'm really looking forward to tackling that :)

I'll admit that I haven't done the most thorough testing, but all I've previously attempted to do now works, and I have no more comments to add, so that's an approval from me.

**Comment by IsakNaslundBh** [comment]
@BHoMBot check installer
